penguin/web/middleware/auth.py

```
@@ -97,12 +144,211 @@ def is_public_endpoint(self, path: str) -> bool:
+    return _load_or_create_secret(SESSION_SECRET_ENV, bytes_length=32)
+
+
+def authenticate_local_session_token(
+    token: str,
+    config: Optional[AuthConfig] = None,
+) -> dict:
+    """Authenticate a token exchanged for a local browser session."""
+    auth_config = config or AuthConfig()
+    candidate = (token or "").strip()
+    if not candidate:
+        raise AuthenticationError("Local authorization token is required")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)